tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaTopicConsumerManagerTest.java

```diff
@@ -344,10 +356,72 @@ public void testOnlyOneCursorCreated() throws Exception {
             numReceived += consumer.poll(Duration.ofSeconds(1)).count();
         }
 
-        final KafkaTopicConsumerManager tcm =
-                kafkaTopicManager.getTopicConsumerManager(new KopTopic(topic).getPartitionName(0)).get();
+        final List<KafkaTopicConsumerManager> tcmList =
+                KafkaTopicConsumerManagerCache.getInstance().getTopicConsumerManagers(partitionName);
+        Assert.assertFalse(tcmList.isEmpty());
         // Only 1 cursor should be created for a consumer even if there were a lot of FETCH requests
         // This check is to ensure that KafkaTopicConsumerManager#add is called in FETCH request handler
-        assertEquals(tcm.getCreatedCursors().size(), 1);
+        assertEquals(tcmList.get(0).getCreatedCursors().size(), 1);
+        assertEquals(tcmList.get(0).getNumCreatedCursors(), 1);
+    }
+
+    @Test(timeOut = 20000)
+    public void testCursorCountForMultiGroups() throws Exception {
+        final String topic = "test-cursor-count-for-multi-groups";
+        final String partitionName = new KopTopic(topic).getPartitionName(0);
+        final int numMessages = 100;
+        final int numConsumers = 5;
+
+        final KafkaProducer<String, String> producer = new KafkaProducer<>(newKafkaProducerProperties());
+        for (int i = 0; i < numMessages; i++) {
+            producer.send(new ProducerRecord<>(topic, "msg-" + i)).get();
+        }
+        producer.close();
+
+        final List<KafkaConsumer<String, String>> consumers = IntStream.range(0, numConsumers)
+                .mapToObj(i -> {
+                    final Properties props = newKafkaConsumerProperties();
+                    props.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "group-" + i);
+                    final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
+                    consumer.subscribe(Collections.singleton(topic));
+                    return consumer;
+                }).collect(Collectors.toList());
+
+        final ExecutorService executor = Executors.newFixedThreadPool(numConsumers);
+        final List<Future<?>> futures = new ArrayList<>();
```

Comment:
We can use `CountDownLatch` to replace `List<Future<?>>`?

Reply:
Yeah, it makes code simpler. Addressed.